tests/integration/os.rs

```diff
@@ -194,3 +194,107 @@ async fn syscalls_cairo1(
+        max_fee,
+        sender_address: sender_address,
+        calldata: create_calldata(contract_address, "test_emit_event", entrypoint_args),
+        version: tx_version,
+        nonce: nonce_manager.next(sender_address),
+    });
+
+    // test_storage_read_write
+    let test_storage_read_write_tx = test_utils::account_invoke_tx(invoke_tx_args! {
+        max_fee,
+        sender_address: sender_address,
+        calldata: create_calldata(contract_address, "test_storage_read_write", &[StarkFelt::TWO, StarkFelt::ONE]),
+        version: tx_version,
+        nonce: nonce_manager.next(sender_address),
+    });
+
+    // test_get_block_hash
+    let test_get_block_hash_tx = test_utils::account_invoke_tx(invoke_tx_args! {
+        max_fee,
+        sender_address: sender_address,
+        calldata: create_calldata(contract_address, "test_get_block_hash", &[stark_felt!(block_context.block_info().block_number.0 - STORED_BLOCK_HASH_BUFFER)]),
+        version: tx_version,
+        nonce: nonce_manager.next(sender_address),
+    });
+
+    // test_send_message_to_l1
+    let to_address = stark_felt!(1234_u16);
+    let payload = vec![stark_felt!(2019_u16), stark_felt!(2020_u16), stark_felt!(2021_u16)];
+    let entrypoint_args = &[vec![to_address, stark_felt!(payload.len() as u64)], payload].concat();
+
+    let test_send_message_to_l1_tx = test_utils::account_invoke_tx(invoke_tx_args! {
+        max_fee,
+        sender_address: sender_address,
+        calldata: create_calldata(contract_address, "test_send_message_to_l1", entrypoint_args),
+        version: tx_version,
+        nonce: nonce_manager.next(sender_address),
+    });
+
+    // test_deploy
+    let test_contract_class_hash = initial_state.cairo1_contracts.get("test_contract").unwrap().class_hash.0;
+    let entrypoint_args = &[
+        test_contract_class_hash, // class hash
+        stark_felt!(255_u8),      // contract_address_salt
+        stark_felt!(2_u8),        // calldata length
+        stark_felt!(3_u8),        // calldata: arg1
+        stark_felt!(3_u8),        // calldata: arg2
+        stark_felt!(0_u8),        // deploy_from_zero
+    ];
+
+    let test_deploy_tx = test_utils::account_invoke_tx(invoke_tx_args! {
+        max_fee,
+        sender_address: sender_address,
+        calldata: create_calldata(contract_address, "test_deploy", entrypoint_args),
+        version: tx_version,
+        nonce: nonce_manager.next(sender_address),
+    });
+
+    let txs = vec![
+        test_emit_event_tx,
+        test_storage_read_write_tx,
+        test_get_block_hash_tx,
+        test_send_message_to_l1_tx,
+        test_deploy_tx,
+    ];
+
+    let _result = execute_txs_and_run_os(
+        initial_state.cached_state,
+        block_context,
+        txs,
+        initial_state.cairo0_compiled_classes,
+        initial_state.cairo1_compiled_classes,
+    )
+    .await
+    .expect("OS run failed");
+}
```

Comment:
Is it possible to create a parameterized test that runs the same txs for different transaction type versions? Would be nice to avoid the duplication and clarify the intent.